### ISSUE TITLE
8bitdo Controller Problem

### ISSUE THREAD

**Issue by capfalcon86** [TeamShinkansen/Hakchi2-CE#335]
Hi there, 

my problem is only when i want to play N64 games on my SNES classic mini. If i play it with my old Wii classic controller there will be no problems and all works fine, but if i play with my 8 bitdo wireless controllers with the retro receiver the d-pad and the left analog stick have the same function and i can´t play any game. But the other games will be work fine. How can i change the controller settings to fix that problem? When I use Retroarch on my PC with Windows the 8bitdo controller will work fine also with N64 games. Only on my SNES classic it will not work

Thx for the Help

**Comment by impeeza** [comment]
I am thinking you are experiencing a Retroarch issue. You could go in Retroarch config and remap the controller. Could be the default mapping is wrong

**Comment by capfalcon86** [comment]
Yes that will be my problem but how exactly do I do it?
I don't have that much experience with it yet

**Comment by impeeza** [comment]
Oki. You can go to libetro oficial site. Is plently of documentation.  By now, if you exit to retroarch menu (by default pressing SELECT+START). And  navigate to: settings, input, there you can map every function of the controllers.

Using "Remap Binds for this core". And "Port X Binds"  you can customize everything on that controller

**Comment by capfalcon86** [comment]
Slowly I think I'm too stupid for that !!!
Every time I change the controller settings, nothing works afterwards. I also searched on libetro but found no instructions on how to do it with the controller.
So it would be nice if someone would help me again. Because no matter what I do, nothing works or nothing changes.

**Comment by impeeza** [comment]
he he he, you're no stupid, is really some times hard to crack.  I use too the same configuration, and in the first was a little bit hard, but in the last version of retroarch, all was setting well by default

**Comment by capfalcon86** [comment]
I am reassured.
I also installed the RetroArch 184 xtreme Ozone from April 27, 2020. But with my 8Bitdo controllers it just doesn't work. Is there a step by step instruction or a video somewhere? Or maybe a configuration file that I just have to insert?

**Comment by capfalcon86** [comment]
Now almost everything works. Only the right analog stick does not want to react when the keys are assigned and neither does L3. What else do I have to consider?

**Comment by impeeza** [comment]
Yes you are all right, the right stick and push buttons of Sticks are no working, but is no Retroarch nor Hakchi fault.  Seems is the Retro Reciver.

For testing I have:
- Wii Classic controller.
- Retro Receiver.
- SN30 Pro+ Controller.
- GBros Adapter (allows convert wireds Gamecube or Wii Controllers to Bbluetooth Switch/X-Input controller)

If I connect the Wii Classic Controller DIRECTLY to the Mini NES Classic Console and run Retroarch, Retroarch detect the «Nintendo Clovercon» connection of the new controller, and when I go to Port 1 Binding the right stick movements are detected as 3 & 4 axis.

Then I connect the Retro Receiver and paired it with the SN30 Pro+ in either mode (D-Inpunt X-Input or Switch) Retroarch detects the new nintendo clovercon (the Retro Receiver should simulate one) but the  right stick is not passed to the mini console.

Then I did try to connect to console the Retro Receiver and paired it to the GBOS adapter with the Wii Classic connected to GBOS and I get the same result:  the Right stick is not passed to the Mini Console nor Retroarch detected the axis.

I am writing to the 8bitdo support for help, you can do it too writing to the email address: support@8bitdo.com

It appear to be a Retro Receiver Issue.

**Comment by impeeza** [comment]
Just Wrote to 8Bitdo support team and they answered:

> Greetings Carlos.
> This is completely normal.
> The left joystick is just like DPAD.
> Has your console been modified?
> We haven't tested non-original system.
> 
> Sincere wishes

So I write back:

Hi, 
First at all, Thanks a lot for answering so quickly,  and for all your help.

Yes, after sending the last e-mail I did realize I didn't provide any of my specs.

I have a SN30 Pro+ with  3.02 firmware.
Retro Receiver with 1.5 beta 1 firmware.
my Mini NES Classic console have Hakchi2 CE 3.8.0 using Retro Arch 1.84

My point is:

If Retro Receiver when connected to Mini NES Classic Console, using the Wiimote port connector "emulate" (or "simulate") a original MINI NES Classic controller, the controller only have a D-Pad and four buttons.

But the Mini NES Classic Console support the Wii Classic Controller (and others Wii (U) controllers).
More information about all the officially supported controllers can be found on the URL:
https://en-americas-support.nintendo.com/app/answers/detail/a_id/17341/~/which-controllers-are-compatible-with-the-nes-classic-edition
which have the next functions:

![image](https://user-images.githubusercontent.com/41979604/80929974-d2782280-8d75-11ea-8aa4-b22ade6f8db0.png)

If Retro Receiver emulate a Wii Classic Controller, it should emulate all of his functions and buttons.  So if the Retro Receiver gets from the Paired Controller, 2 sticks, buttons and D-Pad  they should be mapped to the correspondent ones on the emulated Wii Classic Controller.

Actually, the SN30 Pro+ paired to the Retro Receiver, is mapped in the next way:

![image](https://user-images.githubusercontent.com/41979604/80929986-e4f25c00-8d75-11ea-817a-8043217d88d9.png)

What I asking for is the Left and Right stick (and his buttons) of SN30 Pro+ (or any other controller) paired to the Retro Receiver will be mapped to the corresponding functions on the emulated Wii Classic Controller.  Making the Retro Receiver even more compatible to the Mini NES Classic Console.


Again thanks a lot for your effort.


Let's Waiting the answer.

**Comment by KMFDManic** [comment]
@impeeza A few fixes went into RetroArch Updates to help with 8bitdo, as well as an hmod to help with some easier D Pad N64 Game Mappings.  You still have any issues?

**Comment by impeeza** [comment]
I will test it tomorrow. Thanks for your response

**Comment by impeeza** [comment]
Yes Some button works now, but the right stick still no work, but is a 8BitDo problem. thanks for your effort.

**Comment by KMFDManic** [comment]
@impeeza Have you attempted to map it via RetroArch Settings, Input, Binds, Port 1?  Note, this is global and affects ALL Cores!  You can default bind or tap start to debind.

**Comment by impeeza** [comment]
Doesn't work, but us because the 8BitDo Retro Receivers for NES/SNES only give to the console a d-pad, and 10 butons (x, y, a, b, r, l, zr, zl, start & Select) and no more.  Is like you conect an original snes mini controller, retroarch can not receive Analog stick because the controller do not have analog hardware.

If I connect the 8BitDo controller directly by usb cable, or using Bluetooth adapter ALL buttons and axis works perfect because is hakchi whom make the comunication with controller no the 8BitDo Retro Receiver.  Sumarizing the problem is on the retro receiver firmware and only 8BitDo can solve it.


An extra test I made:
Connecting the 8BitDo Retro Receiver no to frontal port, but via USB cable, retroarch recognize it as a Xbox 360 pad but the buttons are mixed, think 8bitdo receiver emulates a «old» 360 controller firmware
